sotodlib/core/flagman.py

```diff
@@ -0,0 +1,163 @@
+import numpy as np
+from functools import reduce
+
+from so3g.proj import Ranges, RangesMatrix
+from . import AxisManager
+
+class FlagManager(AxisManager):
+    """An extension of the AxisManager class to make functions 
+    more specifically associated with cuts and flags.
+    
+    FlagManager only expects to have items that are mapped to the 
+    detector axis, the sample axis, or both. 
+    
+    Detector Flags can be passed as bitmasks or boolean arrays. To match with
+    Ranges and RangesMatrix, the default is False and the exceptions
+    are True
+    """
+    
+    def __init__(self, det_axis, samp_axis):
+        super().__init__(det_axis, samp_axis)
+                
+        if not 'dets' in self._axes:
+            raise ValueError('FlagManagers require a dets axis')
+        if not 'samps' in self._axes:
+            raise ValueError('FlagManagers require a samps axis')
```

Comment:
I think the issue is with my docstring, which I'll update. I expect the `FlagManager` to always be created with both axes (just like I always expect a TOD that has a samps axis to also have a detector axis, are there times where we wouldn't have that _and_ want flags?) The individual flags in the manager can have any combination of those axes.

I had left out the "exclusivity" checks just in case someone wanted to do something "stupid" that wasn't going to interfere with the functioning of the manager.